### PR TITLE
Update List storage method 

### PR DESCRIPTION
**Pull request by sashah2**
# Summary
Add options to `List` storage method and integrate Delimiter with prefixes.

### CODE REVIEW

File: lib/s3/storage.go [hunk 1]
```diff
@@ -67,7 +87,15 @@ func (s *Storage) List(path string) ([]string, error) {
 		}
 	}
 
-	return result, nil
+	return result
+}
+
+func prefixes(result []string, res *s3.ListObjectsOutput) []string {
```
Comment (madm2n) [inline lib/s3/storage.go]: I think we can make this function more `pure` by doing something like:
```go
func prefixes(res *s3.ListObjectsOutput) []string {
	result := make([]string, 0)

	for _, prefix := range res.CommonPrefixes {
		result = append(result, path.Base(*prefix.Prefix))
	}

	return result
}
```
Then we can do something like this in the `List` function:
```go
result = append(result, prefixes(res)...)
```

**Review by madm2n — COMMENTED** [review]
Good job!

Just one small suggestion.

File: lib/s3/storage.go (hunk 1 above)
Comment (sashah2) [reply lib/s3/storage.go]: Perfect idea! Done.

File: lib/s3/storage.go [hunk 2]
```diff
@@ -38,20 +39,39 @@ type Storage struct {
...
+			}
+
+			return true
+		},
+	)
+
+	return result, err
+}
+
+func contents(result []string, res *s3.ListObjectsOutput) []string {
```
Comment (protsack-stephan) [inline lib/s3/storage.go]: Probably should make this function more `pure` as well to make sure it's consistent. I mean:
```go
func contents(res *s3.ListObjectsOutput) []string
```
Then in list function:
```go
result = append(result, contents(res)...)
```

Comment (sashah2) [reply lib/s3/storage.go]: done.

**Comment by protsack-stephan** [comment]
Good job, feel free to merge!

**Pull request merged**